scanner/src/main/kotlin/ScanStorage.kt

```diff
@@ -39,17 +39,24 @@ sealed interface ScanStorageReader
  */
 interface PackageBasedScanStorageReader : ScanStorageReader {
     /**
-     * Read all [ScanResult]s for the provided [package][pkg]. The results have to match the
-     * [provenance][KnownProvenance.matches] of the package and can optionally be filtered by the provided
-     * [scannerMatcher]. The results are converted to a [NestedProvenanceScanResult] using the provided
+     * Read all [ScanResult]s for the provided [package][pkg]. The package scan results are converted to a
+     * [NestedProvenanceScanResult] using the provided [nestedProvenance].
+     *
+     * Throws a [ScanStorageException] if an error occurs while reading from the storage.
+     */
+    fun read(pkg: Package, nestedProvenance: NestedProvenance): List<NestedProvenanceScanResult>
```

Comment:
> With this change it is no more possible to obtain scan results by identifiers, without provenance.

I'm having difficulties to understand this reason, because all this commit does is to add back a function to the interface that *also* takes a `NestedProvenance`. So how does getting this function back help to read scan results *without* provenance?

Or phrased differently, this code in the `helper-cli`

    scanResultsStorage.read(Package.EMPTY.copy(id = packageId)).getOrThrow()

stays the same before and after this change. What magic does it make work, and why was it broken before?

Reply:
This code filters by provenance:
```
    override fun readInternal(pkg: Package, scannerMatcher: ScannerMatcher?): Result<List<ScanResult>> =
        readForId(pkg.id).map { results -> results.filter { it.provenance.matches(pkg) } }
```
The code before didn't:
```
    override fun readInternal(pkg: Package): Result<List<ScanResult>> {
        val path = storagePath(pkg.id)

        return runCatching {
            backend.read(path).use { input ->
                yamlMapper.readValue<ScanResultContainer>(input).results
            }
        }.recoverCatching {
            // If the file cannot be found it means no scan results have been stored, yet.
            if (it is FileNotFoundException) return Result.success(emptyList())

            val message = "Could not read scan results for '${pkg.id.toCoordinates()}' from path '$path': " +
                it.collectMessages()

            logger.info { message }

            throw ScanStorageException(message)
        }
    }
```